File: minimum_number_of_arrows_to_burst_ballons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:

        # first sort the given points according to starting of the points
        points_sorted = sorted(points, key = lambda x: x[0])
        
        # first we take the first point as common interval
        common = points_sorted[0]      

        # this is a function that will find common interval between two points
        def limit_finder(point1,point2):
            lower_limit = max(point1[0],point2[0])
            higher_limit = min(point1[1],point2[1])
            return [lower_limit,higher_limit]

        # first we initiate the first arrow
        arrow = 1

        # run a loop from second point
        for i in range(1,len(points_sorted)): 
            
            # if  current point lower limit is less than higher limit of common interval
            if points_sorted[i][0] <= common[1]:
                # we find the new common interval between them
                common = limit_finder(common,points_sorted[i])
            else:
                # when current point outside of common interval
                arrow += 1 # we need another arrow
                common = points_sorted[i] # new common interval considered
        return arrow

File: minimum_number_of_arrows_to_burst_ballons.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShots = function(points) {
    const sortedPoints = points.sort((a,b) => a[0] - b[0]);
    console.log(sortedPoints);
    
    const commonIntervalFinder = (point1, point2) => {
        const lowerLimit = Math.max(point1[0],point2[0]);
        const higherLimit = Math.min(point1[1],point2[1]);
        return [lowerLimit,higherLimit];
    };

    let commonInterval = sortedPoints[0];
    let arrow = 1;
    for (let i=1; i<sortedPoints.length; i++){
        if (sortedPoints[i][0] <= commonInterval[1]){
            commonInterval = commonIntervalFinder(commonInterval, sortedPoints[i]);
        }else{
            arrow++;
            commonInterval = sortedPoints[i];
        }
    }    
    return arrow;
};